staging/src/k8s.io/apiserver/pkg/cel/types.go

```diff
@@ -452,6 +452,10 @@ func (rt *DeclTypeProvider) FindFieldType(typeName, fieldName string) (*ref.Fiel
 	}
 
 	f, found := st.Fields[fieldName]
+	if !found && celReservedSymbols.Has(fieldName) {
```

Comment:
this will enable accepting / compiling expressions a 1.30 API server will choke on, right? don't we need to accept this from storage for a release before we accept it in new expressions?

Reply:
https://github.com/kubernetes/kubernetes/blob/e740f8340eedc89baccd120329b454a860385e2d/staging/src/k8s.io/apiserver/pkg/cel/environment/environment.go#L248 looks like a good candidate location to plumb a option in from the environment to enable this change only for the correct environments.